api/v1beta2/moduletemplate_types.go

```diff
@@ -118,7 +118,7 @@ type CustomStateCheck struct {
 	MappedState State `json:"mappedState" yaml:"mappedState"`
 }
 
-func (m *ModuleTemplate) GetDescriptor() (*Descriptor, error) {
+func (m *ModuleTemplate) GetDescriptor(useCache bool) (*Descriptor, error) {
```

Comment:
I suggest change this function to read only mode but no `SetDescToCache`, then we can git rid of this `useCache`, by your change, it makes the logic no so clear.

So for this function:
1. read from cache
2. if no result from cache, fallback to reading from `Descriptor.Raw`

Reply:
Then you only need to set cache in this [condition](https://github.com/kyma-project/lifecycle-manager/pull/916/files#diff-fee2b3ec33be5b4b5a1396a28a16914df3bd0c257297d091d3174c900976e02cR48) for non remote module template

Reply:
By this refactoring, I believe you can further clean up that  `IsRemoteModuleTemplate` field you introduced in `ModuleTemplateTO`

Reply:
I need to save the descriptor to cache in `CheckValidTemplateUpdate` but I don't have access to the module so either I call this function directly from the `GetTemplates` and pass if the module template is remote as an argument or keep it in the `ModuleTemplateTO` (current implementation after I made the GetDescriptor as Read-only). WDYT?